jib-core/src/main/java/com/google/cloud/tools/jib/ncache/CacheWriteEntry.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2018 Google LLC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.ncache;
+
+import com.google.cloud.tools.jib.blob.Blob;
+import com.google.cloud.tools.jib.image.DescriptorDigest;
+import java.util.Optional;
+
+/** Represents layer data to write to the cache. <b>Implementations must be immutable.</b> */
+public interface CacheWriteEntry {
+
+  /**
+   * Gets the {@link Blob} to write as the layer contents.
+   *
+   * @return the layer {@link Blob}
+   */
+  Blob getLayerBlob();
+
+  /**
+   * Gets the optional selector digest to also reference this layer data.
```

Comment:
It is a separate hash to use to reference a layer. We'll have it be a hash of the layer entries in our storage engine implementation.

Reply:
Should it be optional? Are we implying that CacheStorage will have a default implementation to calculate this? Who would and wouldn't return null here?

Reply:
The selector digest will be calculated at a higher level above the `CacheStorage`. Application layers will have selector (since they have layer entries), whereas base image layers will not.

Reply:
Okay, so this is a secondary identifier that can be generated by a cache implementer? Can we add a little more detail to the javadoc?

```
Get an optional selector digest to also reference this layer data. A selector digest may be a secondary identifier for a layer that is distinct from the default layer digest. Useful as an inexpensive alternative reference to a layer rather than going through the motions of calculating the primary layer digest (sha256 of compressed tar)
```